webhook_server_container/libs/github_api.py

```diff
@@ -1629,3 +1696,49 @@ def is_check_run_in_progress(self, check_run):
             if run.name == check_run and run.status == IN_PROGRESS_STR:
                 return True
         return False
+
+    def _run_sonarqube(self):
+        if not self.sonarqube_project_key:
+            return False
+
+        with self._clone_repository(path_suffix=f"sonarqube-{shortuuid.uuid()}"):
+            self.app.logger.info(f"{self.log_prefix} Current directory: {os.getcwd()}")
+            if not self._checkout_pull_request():
+                return False
+
+            self.set_sonarqube_in_progress()
+            target_url = (
+                f"{self.sonarqube_url}/dashboard?id={self.sonarqube_project_key}"
+            )
+            cmd = (
+                f"/sonar-scanner-cli/bin/sonar-scanner -Dsonar.projectKey={self.sonarqube_project_key} "
+                f"-Dsonar.sources=. -Dsonar.host.url={self.sonarqube_url} "
+                f"-Dsonar.token={self.sonarqube_token}"
+            )
+            if run_command(command=cmd, log_prefix=self.log_prefix)[0]:
+                project_status = self.sonarqube_api.qualitygates.request(
+                    path="api/qualitygates/project_status",
+                    params={"projectKey": self.sonarqube_project_key},
+                ).json()
+                if project_status["projectStatus"]["status"]:
+                    return self.set_sonarqube_success(target_url=target_url)
+                else:
+                    return self.set_sonarqube_failure(target_url=target_url)
+
+            else:
+                return self.set_sonarqube_failure(target_url=target_url)
+
+    async def _run_check_runs_async(self):
+        async def _run_check_run(check_run):
+            return check_run()
+
+        task1 = asyncio.create_task(_run_check_run(check_run=self._run_sonarqube))
+        task2 = asyncio.create_task(_run_check_run(check_run=self._run_tox))
+        task3 = asyncio.create_task(
+            _run_check_run(check_run=self._install_python_module)
+        )
+        task4 = asyncio.create_task(_run_check_run(check_run=self._build_container))
```

Comment:
save in list and then for task in tasks:. await...?

Reply:
once it will work as expected, I will make it nicer